xskillscore/tests/test_np_deterministic.py

```diff
@@ -68,6 +70,38 @@ def test_pearson_r_nd(a, b):
     assert_allclose(actual, expected)
 
 
+def test_r2_nd(a, b):
+    axis = 0
+    expected = np.squeeze(a[0, :, :]).copy()
+    for i in range(np.shape(a)[1]):
+        for j in range(np.shape(a)[2]):
+            _a = a[:, i, j]
+            _b = b[:, i, j]
+            expected[i, j] = r2_score(_a, _b)
+    actual = _r2(a, b, weights, axis, skipna)
+    assert_allclose(actual, expected)
+
+    axis = 1
+    expected = np.squeeze(a[:, 0, :]).copy()
+    for i in range(np.shape(a)[0]):
+        for j in range(np.shape(a)[2]):
+            _a = a[i, :, j]
+            _b = b[i, :, j]
+            expected[i, j] = r2_score(_a, _b)
+    actual = _r2(a, b, weights, axis, skipna)
+    assert_allclose(actual, expected)
+
+    axis = 2
+    expected = np.squeeze(a[:, :, 0]).copy()
+    for i in range(np.shape(a)[0]):
+        for j in range(np.shape(a)[1]):
+            _a = a[i, j, :]
+            _b = b[i, j, :]
+            expected[i, j] = r2_score(_a, _b)
+    actual = _r2(a, b, weights, axis, skipna)
+    assert_allclose(actual, expected)
```

Comment:
Would like to see a simple definitive/hand-checked test rather than using a, b which are random arrays. Inspires more confidence that way.

Reply:
e.g. reproducing:
https://www.bing.com/videos/search?q=coefficient+determination+calculation+example&view=detail&mid=905D89D54D3AB6959B47905D89D54D3AB6959B47&FORM=VIRE

Reply:
You can verify the numpy code here:
```
import numpy as np
from sklearn.metrics import r2_score

obs = np.array((3, -0.5, 2, 7))
fct = np.array((2.5, 0.0, 2, 8))
squared_error = (obs - fct) ** 2
r_num = np.sum(squared_error)
obs_mean = np.mean(obs)
obs_anom = obs - obs_mean
obs_anom_squared = obs_anom ** 2
r_den = np.sum(obs_anom_squared)
actual = 1 - (r_num / r_den)
expected = r2_score(obs, fct)

weights = np.array((0.75, 0.25, 0.25, 0.75))
squared_error = weights * ((obs - fct) ** 2)
r_num = np.sum(squared_error)
obs_mean = np.sum(obs * weights) / np.sum(weights)
obs_anom = obs - obs_mean
obs_anom_squared = weights * (obs_anom ** 2)
r_den = np.sum(obs_anom_squared)
actual = 1 - (r_num / r_den)
expected = r2_score(obs, fct, sample_weight=weights)
```

and our implementation:
```
$ git clone https://github.com/raybellwaves/xskillscore.git
$ cd xskillscore
$ git checkout add-r2
$ conda env create -f ci/requirements-py36.yml
$ conda activate xskillscore-dev
$ conda install -c conda-forge ipython -y
$ ipython

import xarray as xr
import numpy as np
import xskillscore as xs
from sklearn.metrics import r2_score

_a = np.array((3, -0.5, 2, 7))
_b = np.array((2.5, 0.0, 2, 8))

a = xr.DataArray(_a)
b = xr.DataArray(_b)

actual = xs.r2(a, b, 'dim_0')
expected = r2_score(_a, _b)

_weights = np.array((0.75, 0.25, 0.25, 0.75))
weights = xr.DataArray(_weights)

actual = xs.r2(a, b, 'dim_0', weights=weights)
expected = r2_score(_a, _b, sample_weight=_weights)
```

as for how accurate it is in sklearn or how well tested i'm not sure. But i'm guessing it's pretty good.